x/incentives/keeper/distribute_test.go

```diff
@@ -1131,3 +1125,253 @@ func (s *KeeperTestSuite) IncentivizeInternalGauge(poolIds []uint64, epochDurati
+			clPool := s.PrepareConcentratedPool()
+
+			// create 3 internal Gauge
+			internalGauges := s.setupNoLockInternalGauge(clPool.GetId(), uint64(6)) // gauge Id = 2,3,4,5,6,7
+
+			// create non-perp internal Gauge
+			s.CreateNoLockExternalGauges(clPool.GetId(), sdk.NewCoins(), s.TestAccs[1], uint64(2)) // gaugeid= 8
+
+			// create perp group gauge
+			_, err := s.App.IncentivesKeeper.CreateGroupGauge(s.Ctx, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100_000_000))), uint64(1), s.TestAccs[1], internalGauges[:3], lockuptypes.ByGroup, types.Evenly) // gauge id = 2,3,4
+			s.Require().NoError(err)
+
+			// create non-perp group gauge
+			_, err = s.App.IncentivesKeeper.CreateGroupGauge(s.Ctx, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(100_000_000))), uint64(2), s.TestAccs[1], internalGauges[len(internalGauges)-3:], lockuptypes.ByGroup, types.Evenly) // gauge id = 5,6,7
+			s.Require().NoError(err)
+
+			// Call Testing function
+			err = s.App.IncentivesKeeper.AllocateAcrossGauges(s.Ctx)
+			if tc.expectError {
+				s.Require().Error(err)
+			} else {
+				s.Require().NoError(err)
+
+				groupGaugePostAllocate, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, tc.GroupGauge.GroupGaugeId)
+				s.Require().NoError(err)
+
+				s.Require().Equal(groupGaugePostAllocate.DistributedCoins, tc.expectedAllocationPerGroupGauge)
+
+				for _, gauge := range tc.GroupGauge.InternalIds {
+					internalGauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, gauge)
+					s.Require().NoError(err)
+
+					s.Require().Equal(internalGauge.Coins, tc.expectedAllocationPerInternalGauge)
+				}
+			}
+		})
+	}
+}
+
+func (s *KeeperTestSuite) WithBaseCaseDifferentCoins(baseCase GroupGaugeCreationFields, newCoins sdk.Coins) GroupGaugeCreationFields {
+	baseCase.coins = newCoins
+	return baseCase
+}
+
+func (s *KeeperTestSuite) WithBaseCaseDifferentEpochPaidOver(baseCase GroupGaugeCreationFields, numEpochPaidOver uint64) GroupGaugeCreationFields {
+	baseCase.numEpochPaidOver = numEpochPaidOver
+	return baseCase
+}
+
+func (s *KeeperTestSuite) WithBaseCaseDifferentInternalGauges(baseCase GroupGaugeCreationFields, internalGauges []uint64) GroupGaugeCreationFields {
+	baseCase.internalGaugeIds = internalGauges
+	return baseCase
+}
+
+func (s *KeeperTestSuite) TestCreateGroupGaugeAndDistribute() {
+	hundredKUosmo := sdk.NewCoin("uosmo", sdk.NewInt(100_000_000))
+	hundredKUatom := sdk.NewCoin("uatom", sdk.NewInt(100_000_000))
+	fifetyKUosmo := sdk.NewCoin("uosmo", sdk.NewInt(50_000_000))
+	fifetyKUatom := sdk.NewCoin("uatom", sdk.NewInt(50_000_000))
+	twentyfiveKUosmo := sdk.NewCoin("uosmo", sdk.NewInt(25_000_000))
+	twentyfiveKUatom := sdk.NewCoin("uatom", sdk.NewInt(25_000_000))
+
+	baseCase := &GroupGaugeCreationFields{
+		coins:            sdk.NewCoins(hundredKUosmo),
+		numEpochPaidOver: 1,
+		owner:            s.TestAccs[1],
+		internalGaugeIds: []uint64{2, 3, 4, 5},
+	}
+
+	tests := []struct {
```

Comment:
imo the `map[string]struct` structure is much cleaner (e.g. see how it's done [here](https://github.com/osmosis-labs/osmosis/blob/2a9b63336d83ac59cae5a1ab1e14f47eeedb08fc/x/concentrated-liquidity/model/pool_test.go#L588) – no need for name as a field and you can pick up the test case name from the first for loop field on line 1289 below)

Reply:
yea, i personally like writing name field, it just gives me more clarity while writing tests 😅

happy to change tho if we want to stay consistent with rest of the test